### ISSUE TITLE
salt-cloud with ec2 provider fails on non-standard ssh port

### ISSUE THREAD

**Issue by sjwoodr** [saltstack/salt#21434]
When using a non-standard port for SSH, salt-cloud fails to bootstrap a newly created EC2 instance. 

Use case: we run sshd on a port other than 22 in all of our EC2 instances in AWS.  

Reproduce:  specify an AMI in your profile which has sshd running on a port other than 22.  The instance will be created, but the bootstrapping will fail.  

This issue exists in 2014.7.1 as well as in the develop branch.   I have a fix for this tested and ready to be reviewed.  

Version Info:

```
           Salt: 2014.7.1
         Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
 msgpack-python: 0.3.0
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.10
          ioflo: Not Installed
          PyZMQ: 14.0.1
           RAET: Not Installed
            ZMQ: 4.0.4
           Mako: 0.9.1
```

**Comment by rallytime** [comment]
Thanks for this report, and for adding your fix @sjwoodr. We love it when that happens! :)

**Comment by rallytime** [comment]
@sjwoodr Since your pull request has been merged, is this ok to close? Or does more work need to be done here?

**Comment by sjwoodr** [comment]
@rallytime I think we're good to go -- Thank you!

**Comment by rallytime** [comment]
Awesome! Thanks very much! I backported the fix to the 2015.2 branch, so it will be available in 2015.2.0 release.
